storage/pruning/pruningStorer.go

```diff
@@ -260,6 +260,25 @@ func (ps *PruningStorer) GetFromEpoch(key []byte, epoch uint32) ([]byte, error)
 
 }
 
+// SearchFirst will search a given key in all the active persisters, from the newest to the oldest
+func (ps *PruningStorer) SearchFirst(key []byte) ([]byte, error) {
+	ps.lock.Lock()
+	defer ps.lock.Unlock()
+
+	for _, pd := range ps.activePersisters {
```

Comment:
Why not iterating from the most recent persister? Most requests will come from current epoch.